File: Factory/Factory/Objective-C/ViewController.m
//
//  ViewController.m
//  Factory
//
//  Created by LMinh on 8/9/18.
//  Copyright © 2018 LMinh. All rights reserved.
//

#import "ViewController.h"
#import "MonsterFactory.h"

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    Monster *lion = [MonsterFactory createMonsterWithType:MonsterType_Lion];
    Monster *dragon = [MonsterFactory createMonsterWithType:MonsterType_Dragon];
    Monster *wolf = [MonsterFactory createMonsterWithType:MonsterType_Wolf];
    
    [lion appear];
    [dragon appear];
    [wolf appear];
    
    int randomType = arc4random_uniform(3);
    Monster *randomMonster = [MonsterFactory createMonsterWithType:randomType];
    [randomMonster appear];
}

@end

File: Factory/Factory/Swift/ViewController.swift
//
//  ViewController.swift
//  Factory
//
//  Created by LMinh on 8/9/18.
//  Copyright © 2018 LMinh. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        let lion = MonsterFactory.createMonster(type: .Lion)
        let dragon = MonsterFactory.createMonster(type: .Dragon)
        let wolf = MonsterFactory.createMonster(type: .Wolf)
        
        lion.appear()
        dragon.appear()
        wolf.appear()
        
        let randomType = Int(arc4random_uniform(3))
        let randomMonster = MonsterFactory.createMonster(type: MonsterType(rawValue: randomType)!)
        randomMonster.appear()
    }
}

